dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -2517,8 +2518,10 @@ public void deleteRelatedContent(final Contentlet contentlet, final Relationship
             }
         }
 
-        // Refresh the parent
-        refreshNoDeps(contentlet);
+        // Refresh the parent only if the contentlet is not already in the checkin
+        if (contentlet.get(CHECKIN_IN_PROGRESS) == null || !contentlet.getBoolProperty(CHECKIN_IN_PROGRESS)) {
```

Comment:
Could be shortened:
` if (!contentlet.getBoolProperty(CHECKIN_IN_PROGRESS))`

Reply:
good catch done